src/vision/status_quo/barbara_bridges_sync_and_async.md

```diff
@@ -0,0 +1,245 @@
+
+```rust
+fn aggregate(urls: &[Url]) {
+    let data: Vec<Data> =
+        urls
+            .iter()
+            .map(|url| tokio::task::spawn_blocking(move || {
+                tokio::task::spawn(async move {
+                    do_web_request(url).await
+                })
+            }))
+            .collect();
+}
+```
+
+But this isn't really helping, as `spawn` still yields a future. She's getting the same errors.
+
+### Async all the way
+
+She remembers now that this whole drama started because she was converting her `main` function to be `async`. Maybe she doesn't have to bridge between sync and async? She starts digging around in the docs and finds `futures::join_all`. Using that, she can change `aggregate` to be an async function too:
+
+```rust
+async fn aggregate(urls: &[Url]) {
+    let data: Vec<Data> = futures::join_all(
+        urls
+            .iter()
+            .map(|url| do_web_request(url))
+    ).await;
+}
+```
+
+Things are working again now, so she is happy.
+
+### Filtering
+
+Later on, she would like to apply a filter to the aggregation operation. She realizes that if she wants to use the fetched data when doing the filtering, she has to filter the vector after the join has completed; `join_all` doesn't have a way to put a filter into the iterator chain like she wants. She is annoyed, but performance isn't critical, so it's ok.
+
+### And the cycle begins again
+
+Later on, she wants to call `aggregate` from another binary. This one doesn't have an `async main`. This context is deep inside of an iterator chain and was previously entirely synchronous. She realizes it would be a lot of work to change all the intervening stack frames to be `async fn`, rewrite the iterators into streams, etc. She decides to just call `block_on` again, even though it make her nervous.
+
+## 🤔 Frequently Asked Questions
+
+### What are the morals of the story?
+
+* Some projects don't care about max performance and just want things to work.
+    * They would probably be happy with sync but as the most popular libraries for web requests, databases, etc, offer async interfaces, they may still be using async code.
+* There are contexts where you can't easily add an `await`.
+    * For example, inside of an iterator chain.
+    * Big block of existing code.
+* Mixing sync and async code (`block_on`) can cause deadlocks that are really painful to diagnose.
+
+
+### Why did you choose Barbara to tell this story?
+
+* Because Mark (who experienced most of it) is a very experienced Rust developer.
+* Because you could experience this story regardless of language background or being new to Rust.
+
+### How would this story have played out differently for the other characters?
+
+I would expect it would work out fairly similarly, except that the type errors and things might well have been more challenging for people to figure out, assuming they aren't already familiar with Rust.
```

Comment:
```suggestion
I would expect it would work out fairly similarly, except that the type errors and things might well have been more challenging for people to figure out, assuming they aren't already familiar with async Rust.
```

Familiarity with sync rust does not help much when coming to async I believe.